core/src/main/java/org/elasticsearch/indices/breaker/StaticCircuitBreakerService.java

```diff
@@ -26,14 +26,30 @@
 /**
  * Class that returns a breaker that never breaks
  */
-public class NoneCircuitBreakerService extends CircuitBreakerService {
+public class StaticCircuitBreakerService extends CircuitBreakerService {
```

Comment:
You're right. When unit-testing the `*ServerTransport` classes I needed to provide a `CircuitBreakerService` but I need more fine-grained control over the behaviour of the `CircuitBreaker` that is returned (note that in these tests I just need one breaker). In all integration tests (which use Guice) we still have the old logic of returning always a `NoopCircuitBreaker`.

I don't think that we need all the registration stuff in the circuit breaker module but we should instead define more specifically which breaker implementation we want to return for which breaker type. But as I said this is just needed for unit tests, so I would rather provide a more generic implementation that allows this registration (for unit tests) and have `StaticCircuitBreakerService`(or `NoneCircuitBreakerService`) just use it internally.